packages/lab/src/custom-slider/CustomSlider.tsx

```diff
@@ -0,0 +1,208 @@
+      css: sliderCss,
+      window: targetWindow,
+    });
+
+    const [value, setValue] = useControlled({
+      controlled:
+        valueProp === undefined ? undefined : clamp(valueProp, min, max),
+      default: defaultValue,
+      name: "Slider",
+      state: "value",
+    });
+
+    const [tooltipVisible, setTooltipVisible] = useState(false);
+    const [isDragging, setIsDragging] = useState(false);
+    const sliderRef = useRef<HTMLDivElement>(null);
+
+    const calculateThumbPosition = (clientX: number) => {
+      if (sliderRef.current) {
+        const sliderRect = sliderRef.current.getBoundingClientRect();
+        const rawValue =
+          ((clientX - sliderRect.left) / sliderRect.width) * (max - min) + min;
+        const steppedValue = Math.round(rawValue / step) * step;
+        const newValue = Math.min(Math.max(min, steppedValue), max);
+        setValue(newValue);
+        onChange?.(newValue);
+      }
+    };
+
+    const handleMouseMove = (event: MouseEvent) => {
+      calculateThumbPosition(event.clientX);
+    };
+
+    const handleMouseUp = () => {
+      setIsDragging(false);
+      setTooltipVisible(false);
+      document.removeEventListener("mousemove", handleMouseMove);
+      document.removeEventListener("mouseup", handleMouseUp);
+    };
+
+    const handleMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+      calculateThumbPosition(event.clientX);
+      setIsDragging(true);
+      document.addEventListener("mousemove", handleMouseMove);
+      document.addEventListener("mouseup", handleMouseUp);
+    };
+
+    const handleMouseEnter = () => {
+      setTooltipVisible(true);
+    };
+
+    const handleMouseLeave = () => {
+      if (!isDragging) setTooltipVisible(false);
+    };
+
+    const handleFocus = () => setTooltipVisible(true);
+
+    const handleBlur = () => setTooltipVisible(false);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+      let newValue = value;
+      switch (event.key) {
+        case "ArrowUp":
+          newValue = Math.min(value + step, max);
+          break;
+        case "ArrowRight":
+          newValue = Math.min(value + step, max);
+          break;
+        case "ArrowDown":
+          newValue = Math.max(value - step, min);
+          break;
+        case "ArrowLeft":
+          newValue = Math.max(value - step, min);
+          break;
+        case "Home":
+          newValue = min;
+          break;
+        case "End":
+          newValue = max;
+          break;
+      }
+
+      setValue(newValue);
+      onChange?.(newValue);
+    };
+
+    const progressPercentage = ((value - min) / (max - min)) * 100;
+
+    return (
+      <div
+        className={clsx(
+          withBaseName(),
+          withBaseName(`${labelPosition}Labels`),
+          { dragging: isDragging },
+        )}
+        ref={ref}
+        {...rest}
+      >
+        <Text as="label" className={clsx(withBaseName("minLabel"))}>
```

Comment:
```suggestion
        <Text as="label" className={withBaseName("minLabel")}>
```
More of a nit, we don't need to use `clsx` here.